docs/limitations.rst

```diff
@@ -0,0 +1,177 @@
+driver. Set ``HDF5_DO_MPI_FILE_SYNC=1`` to enable this option.
+
+**Cost:** Can cause a significant decrease in write performance as the amount of
+file sync operations performed will likely be more than necessary. Similar to,
+but potentially more efficient than, the ``WRITE_SYNC`` workaround as less
+overall file syncs may be performed in comparision, but still likely more than
+needed.
+
+ROMIO Driver Hint
+"""""""""""""""""
+
+A ROMIO provided hint that will cause the ROMIO driver (in a supported MPI
+library) to add an ``MPI_Barrier()`` call and an additional ``MPI_File_sync()``
+call after each already existing ``MPI_File_sync()`` call within the
+application. In other words, this hint converts each existing
+``MPI_File_sync()`` call into the "sync-barrier-sync" construct. Enable the
+``romio_synchronizing_flush`` hint to use this workaround.
+
+**Cost:** Potentially more efficient that the ``WRITE_SYNC`` and HDF5
+``FILE_SYNC`` workarounds as this will cause the application to use the
+synchronization construct required by UnifyFS everywhere the application already
+intends them to occur (i.e., whenever there is already an ``MPI_File_sync()``).
+However, if (1) any existing ``MPI_File_sync()`` calls are only meant to make
+data visible to the other processes (rather than to avoid potential conflicts)
+or (2) the application contains a mix of lone ``MPI_File_sync()`` calls along
+with the "sync-barrier-sync" construct, then this approach will result in more
+syncs than necessary.
+
+----------
+
+.. _file-lock-label:
+File Locking
+************
+
+UnifyFS not supporting file locks results in some I/O library features to not
+work with UnifyFS.
+
+.. topic:: Atomicity
+
+    ROMIO uses ``fcntl()`` to implement atomicity. It is recommended to disable
+    atomicity when integrating with UnifyFS. To disable, run
+    ``MPI_File_set_atomicity(fh, 0)``.
+
+.. topic:: Data Sieving
+
+    It is recommended to disable data sieving when integrating with UnifyFS.
+    Even with locking support, use of data sieving will drastically increase the
+    time and space overhead within UnifyFS, significantly decreasing application
+    performance. For ROMIO, set the hints ``romio_ds_write disable`` and
+    ``romio_ds_read disable`` to disable data sieving.
+
+.. topic:: Shared File Pointers
+
+    Avoid using shared file pointers in MPI-I/O under UnifyFS as they require
+    file locking to implement.
+    Functions that use shared file pointers include:
+
+    - ``MPI_File_write_shared()``
+    - ``MPI_File_read_shared()``
+    - ``MPI_File_write_ordered()``
+    - ``MPI_File_read_ordered()``
+
+File Locking Workarounds
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+UnifyFS doesn't provide any direct workarounds for anything that requires file
+locking. Simply disable atomicity and data sieving and avoid using shared file
+pointers to get around this.
+
+Once UnifyFS does provide support for file locking to be enabled, only do so in
```

Comment:
"Once ... without it."
replace with:
"In the future, UnifyFS may add support for file locking. However, it is strongly suggested to avoid file locking unless the application cannot run properly without its use."